### ISSUE TITLE
Research: Spectral init robustness to UMAP parameter sweeps

### ISSUE THREAD

**Issue by Trecek** [TalonT-Org/spectral-init#207]
## Research Question

Does spectral initialization provide robustness to UMAP hyperparameter choices
compared to other initialization methods, and do the Rust and Python spectral
implementations remain equivalent across the parameter space?

## Motivation

The 10K MERFISH benchmark (PR #206) established that Rust spectral init is
equivalent to Python at a single parameter configuration. But the real power of
spectral initialization isn't just global structure preservation at one setting —
it's **stability across parameter sweeps**. A good initialization should produce
consistent, high-quality embeddings regardless of whether `n_neighbors` is 5 or
50, or `min_dist` is 0.01 or 0.5. Random and PCA initializations are expected
to show more sensitivity to these choices.

This research aims to demonstrate that empirically on real spatial
transcriptomics data.

### Dataset

MERFISH 10K subset (committed at `tests/visual_eval/merfish_data/`). Stick to
10K for now — scaling to larger subsets is future work.

### Initialization Methods to Compare

| Init | Source | Notes |
|---|---|---|
| Rust spectral | `spectral_init()` from this crate | Our implementation |
| Python spectral | `umap.spectral.spectral_layout()` | Reference ARPACK-based |
| PCA | `umap.UMAP(init='pca')` | Linear DR init |
| Random | `umap.UMAP(init='random')` | Baseline / lower bound |
| t-SNE | `openTSNE` or `sklearn.manifold.TSNE` | Nonlinear alternative (not an init — a separate DR method for comparison) |

For t-SNE, use whichever is the most popular/standard Python package. openTSNE
is preferred if available (supports callbacks, partial embeddings), otherwise
fall back to sklearn's implementation.

### UMAP Parameter Sweep Grid

| Parameter | Values | Default |
|---|---|---|
| `n_neighbors` | 5, 10, 15, 30, 50, 100 | 15 |
| `min_dist` | 0.0, 0.01, 0.1, 0.25, 0.5, 0.8 | 0.1 |
| `metric` | euclidean, cosine | euclidean |

**Sweep strategy:** Vary one parameter at a time from defaults (not full grid —
that would be 72 configs × 4 inits = 288 UMAP runs). So:
- 6 `n_neighbors` values × 4 inits = 24 runs (min_dist=0.1, metric=euclidean)
- 6 `min_dist` values × 4 inits = 24 runs (n_neighbors=15, metric=euclidean)
- 2 `metric` values × 4 inits = 8 runs (n_neighbors=15, min_dist=0.1)
- Plus t-SNE runs at corresponding perplexity values

Total: ~56 UMAP runs + t-SNE runs. At ~15s per run on 10K cells, this is
~15 minutes total.

### Metrics to Measure Per Configuration

**Embedding quality (vs high-dim PCA features):**
- Trustworthiness (k=15)
- kNN preservation rate (k=15)
- Random triplet accuracy

**Stability across sweep:**
- For each init method, compute the **coefficient of variation** (CV = std/mean)
  of each quality metric across the parameter sweep. Lower CV = more robust.
- Pairwise Procrustes disparity between the default-param embedding and each
  swept-param embedding for the same init method (measures how much the
  embedding shape changes)

**Rust-vs-Python spectral drift:**
- For each parameter config, compute Procrustes disparity between the Rust-init
  and Python-init UMAP embeddings. If spectral init is truly robust, this should
  remain small across the entire sweep — no parameter setting should magnify
  differences between the two implementations.

**Spatial preservation (MERFISH-specific):**
- SNA at each configuration (may or may not differentiate at 10K)
- Moran's I at each configuration

### t-SNE Comparison

Run t-SNE with perplexity values roughly matching the UMAP `n_neighbors` sweep:
perplexity ∈ {5, 10, 15, 30, 50}. Compute the same quality metrics. This
provides an external reference for what "good DR" looks like on this data,
independent of UMAP entirely.

### Expected Findings

1. **Spectral init (both Rust and Python) should show lowest CV** across the
   parameter sweep — most stable quality metrics regardless of UMAP settings
2. **Random init should show highest CV** — most sensitive to parameter choices
3. **PCA init should be intermediate** — reasonably stable but less than spectral
4. **Rust and Python spectral should track each other** — Procrustes between
   them should remain small (<0.05) across all parameter settings
5. **t-SNE should show different sensitivity patterns** — since it has its own
   optimization, not just initialization

### Output

- A results CSV with columns: `init_method`, `param_swept`, `param_value`,
  `trustworthiness`, `knn_preservation`, `triplet_accuracy`, `sna`, `morans_i`,
  `procrustes_vs_default`, `procrustes_rust_vs_python`
- Summary tables showing CV per init method per metric
- Plots:
  - Line plots: metric vs parameter value, one line per init method
  - Bar chart: CV comparison across init methods
  - Heatmap: Rust-vs-Python Procrustes across the full parameter space
- Research report in `research/` folder

## Execution

This issue is designed for the local research recipe
(`.autoskillit/recipes/research.yaml`):

```
task: "Spectral initialization robustness to UMAP parameter sweeps on MERFISH 10K"
source_dir: /home/talon/projects/spectral-init
base_branch: main
setup_phases: true
```

Phase 1 will scope and produce an experiment plan for human review.
Phase 2 (after approval) implements, runs, and writes the report.

## Notes

- This is likely to generate follow-up research questions based on what we find
- Future work: repeat at 100K+ to see if sensitivity patterns change with scale
- The t-SNE comparison is informational — we're not claiming UMAP > t-SNE,
  just providing context for what robustness looks like across methods
